www/front_src/src/Resources/Graph/Performance/ExportableGraphWithTimeline/index.tsx

```diff
@@ -202,73 +210,64 @@ const ExportablePerformanceGraphWithTimeline = ({
-                getIsModalOpened={getIsModalOpened}
-                performanceGraphRef={
-                  performanceGraphRef as unknown as MutableRefObject<HTMLDivElement | null>
-                }
-                resourceName={resource?.name as string}
-                resourceParentName={resource?.parent?.name}
-                resourceType={resource?.type}
-                timeline={timeline}
-              />
-            )
+            <MemoizedGraphActions
+              customTimePeriod={customTimePeriod}
+              getIsModalOpened={getIsModalOpened}
+              open={interactWithGraph}
+              performanceGraphRef={
+                performanceGraphRef as unknown as MutableRefObject<HTMLDivElement | null>
+              }
+              resourceName={resource?.name as string}
+              resourceParentName={resource?.parent?.name}
+              resourceType={resource?.type}
+              timeline={timeline}
+            />
           }
           graphHeight={graphHeight}
-          isEditAnomalyDetectionDataDialogOpen={
-            isEditAnomalyDetectionDataDialogOpen
-          }
+          interactWithGraph={interactWithGraph}
           isInViewport={isInViewport}
           limitLegendRows={limitLegendRows}
           modal={
-            isOpenModalAD && (
-              <EditAnomalyDetectionDataDialog
-                isOpen={isOpenModalAD}
-                setIsOpen={setIsOpenModalAD}
-              >
-                {({
-                  factorsData,
-                  getFactors,
-                  openModalConfirmation,
-                  isEnvelopeResizingCanceled,
-                  isResizeEnvelope,
-                  setIsResizeEnvelope,
-                }): JSX.Element => (
-                  <>
-                    {factorsData && (
-                      <ExportablePerformanceGraphWithTimeline
-                        isEditAnomalyDetectionDataDialogOpen
-                        graphHeight={180}
-                        resizeEnvelopeData={factorsData}
-                        resource={resource}
-                      />
-                    )}
-                    {getFactors && details && details?.sensitivity && (
-                      <EditAnomalyDetectionDataDialog.Slider
-                        details={details}
-                        isEnvelopeResizingCanceled={isEnvelopeResizingCanceled}
-                        isResizeEnvelope={isResizeEnvelope}
-                        openModalConfirmation={openModalConfirmation}
-                        sendFactors={getFactors}
-                        sendReloadGraphPerformance={sendReloadGraphPerformance}
-                        sensitivity={details.sensitivity}
-                        setIsResizeEnvelope={setIsResizeEnvelope}
-                      />
-                    )}
-                  </>
-                )}
-              </EditAnomalyDetectionDataDialog>
-            )
+            <EditAnomalyDetectionDataDialog
```

Comment:
I think you can remove the `modal` prop and extract it outside ExportableGraphWithTimeline and use graphActions instead